app/controllers/admin/edition_images_controller.rb

```diff
@@ -26,9 +26,10 @@ def create
     new_image = @edition.images.build
     new_image.build_image_data(image_params["image_data"])
 
-    if @edition.save
+    if @edition.save && new_image.valid?
       redirect_to admin_edition_image_preview_path(@edition, new_image.id)
     else
+      @edition.reload
```

Comment:
I'm not entirely sure if/why this `@edition.reload` call is needed? 🤔 

I would have expected that to actually reset the `@edition.errors` state and disconnect the `new_image` object from that edition.

Does it work without this line?

Reply:
As per our Slack conversation: This line was to remove the `new_image` object from `@edition`. Without this, the index page errors as it attempts to render the `new_image` object in the "Uploaded Images" section. 

We agreed to change this to `@edition.images.delete(@new_image)`, to avoid reloading the entire edition which could be costly for large editions with lots of images and to include a comment to explain why this line is needed.